File: app/src/main/java/org/ole/planet/myplanet/model/SourceFile.java
package org.ole.planet.myplanet.model;

import java.util.ArrayList;
import java.util.List;

public class SourceFile {
    private String mTitle;
    private String mResType;
    private String mURL;
    private String mDate;

    private List mSubjects = new ArrayList();
    private List mLevels = new ArrayList();

    // Default Constructor
    public SourceFile() {
    }

    public SourceFile(String title, String resType, String URL, String date, List subjects, List levels) {
        mTitle = title;
        mResType = resType;
        mURL = URL;
        mDate = date;
        mSubjects = subjects;
        mLevels = levels;
    }

    public String getTitle() {
        return mTitle;
    }

    public void setTitle(String title) {
        mTitle = title;
    }

    public String getDate() {
        return mDate;
    }

    public void setDate(String date) {
        mDate = date;
    }

}

File: app/src/main/java/org/ole/planet/myplanet/model/SourceFile.kt
package org.ole.planet.myplanet.model

class SourceFile {
    var title: String? = null
    private var mResType: String? = null
    private var mURL: String? = null
    var date: String? = null
    private var mSubjects: List<*> = ArrayList<Any?>()
    private var mLevels: List<*> = ArrayList<Any?>()

    constructor()
    constructor(title: String?, resType: String?, URL: String?, date: String?, subjects: List<*>, levels: List<*>) {
        this.title = title
        mResType = resType
        mURL = URL
        this.date = date
        mSubjects = subjects
        mLevels = levels
    }
}
